text/0000-namespace-syscalls.md

```diff
@@ -0,0 +1,75 @@
+- Feature Name: namespace-syscalls
+- Start Date: 2016-11-23
+- RFC PR: (leave this empty)
+- Redox Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+A namespace is designed to implement the following with one abstraction:
+- `cap_enter`, by default
+- `chroot`, by allowing a filter on `file:`
+- [OS-level virtualization](https://en.wikipedia.org/wiki/Operating-system-level_virtualization) such as [FreeBSD-style Jails](https://en.wikipedia.org/wiki/FreeBSD_jail) or [Illumos-style Zones](https://en.wikipedia.org/wiki/Solaris_Containers), with more complex filtering of scheme access
+
+It achieves this with the addition of three syscalls:
+- `getns`, which gets the current namespace
+- `mkns`, which creates a new namespace
+- `setns`, which switches namespaces
+
+# Motivation
+[motivation]: #motivation
+
+Why are we doing this? What use cases does it support? What is the expected outcome?
+
+# Detailed design
```

Comment:
After reading this section, the intended semantics are still pretty vague to me. What does it mean to be in a given namespace? I assume that the scheme is aware of the namespace of the caller, right? Is there a hierarchy?

Also, what's the argument to `mkns`?  What's an _empty_ namespace?

Does this affect only processes or also threads? Does the context need to remember all the namespaces that it has already had?